server/src/controllers/csv.js

```diff
@@ -0,0 +1,174 @@
+                { headers: { "Content-type": "application/json" } }
+              )
+            ).data.url;
+
+            await axios.put(url, req.file, {
+              headers: {
+                "Content-Type": req.file.mimetype,
+              },
+            });
+          } catch (err) {
+            // console.log("Error happened");
+            console.log("Error uploading to cloud:", err);
+          }
+
+          // Create a new entry in Files table to manage all files a user has uploaded
+          await db.Files.create({
+            name: filename,
+            userId: req.currentUser.id,
+            type: req.file.mimetype,
+          });
+
+          // Parse through the csv and find what fields does it contain
+          fs.createReadStream(`./prospects/${filename}`)
+            .pipe(csv())
+            .on("data", (data) => results.push(data))
+            .on("end", () => {
+              // Check if the csv is empty
+              if (!(results.length > 0)) {
+                return res
+                  .status(400)
+                  .send({ errors: [{ message: "Empty CSV file" }] });
+              }
+
+              const properties = Object.keys(results[0]);
+
+              // Get the attributes of the Prospect Table 
+              const attributes = (Object.keys(db.sequelize.models.Prospect.rawAttributes))
+                .filter(attr => {
+                  return ['createdAt', 'updatedAt', 'status', 'id'].indexOf(attr) === -1
+                })
+              console.log(attributes)
+
+
+              // Set up a cookie equal to the file name for reference
+              req.session = {
+                ...req.session,
+                filename: filename
+              }
+
+              // Send message to client
+              return res.send({
+                message: "Files uploaded successfully!",
+                properties,
+                attributes
+              });
+            });
+        } catch (err) {
+          console.log("Caught:", err);
+          return res
+            .status(401)
+            .send({ errors: [{ message: "Error analyzing files, try again later" }] });
+        }
+      }
+    });
+  }
+
+  // The controller that handles adding prospects to the DB
+  static addProspects = async (req, res) => {
+    // Name of the file in which the data is stored
+    const filename = req.session.filename
+
+    // The mapped attributes that the user selected
+    const mappedAttributes = req.body.mappedAttributes
+    const results = []
+
+    // Read the csv and create new prospects
+    try {
+      fs.createReadStream(`./prospects/${filename}`)
+        .pipe(csv())
+        .on("data", (data) => results.push(data))
+        .on("end", async () => {
+          if (!(results.length > 0)) {
+            return res.send(400).send({ errors: [{ message: "Empty csv file" }] })
+          } else {
+            await db.Prospect.bulkCreate(results.map((result) => {
+              return {
```

Comment:
Do you add the correct owner (userId) to this prospect?

Reply:
The db instance I had running locally didn't have up-to-date model definitions at time. I was waiting on dev to be approved. I would merge dev into mine and add a new commits to mine now.